### PR TITLE
Including the "metadata" field into the notification payload

### PR DESCRIPTION
**Pull request by acierto**
## Hey, I just made a Pull Request!

Implementation of this proposal: https://github.com/backstage/backstage/pull/24562 

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
ping @backstage/notifications-maintainers @drodil

File: plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts [hunk 1]
```diff
@@ -165,6 +172,15 @@ export class DatabaseNotificationsStore implements NotificationsStore {
       q.where('user', user).orWhereNull('user');
     });
 
+    Object.keys(options.metadata ?? {}).forEach(key => {
+      query.whereJsonPath(
```
Comment (mareklibra) [inline plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts:176]: To accept this, we must extend our test matrix for all supported databases in the `router.test.ts`, so far it is with `sqlite3` only.
We can mimic what other tests are doing with the `TestDatabases` from `@backstage/backend-test-utils`

Comment (freben) [reply plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts:176]: oh for sure, all database related code should be covered by TestDatabases (even if it's only for a subset of all possible databases). Do not use `DatabaseManager` directly in tests.

Comment (freben) [reply plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts:176]: But `DatabaseNotificationsStore.test.ts` [already uses that](https://github.com/backstage/backstage/blob/9b3399e06ec2ece812446cf0503eb443f65d07a4/plugins/notifications-backend/src/database/DatabaseNotificationsStore.test.ts#L153), right?

Comment (mareklibra) [reply plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts:176]: Right, I was confused with the `router.test.ts` which can stay with one DB only as long as we do sufficient testing of the DatabseNotificationsStore.

@benjdlambert  can you please add more filtering cases for Json queries there? So far it's just full equality.

Comment (benjdlambert) [reply plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts:176]: @mareklibra I think you meant to ping @acierto 😅

Comment (acierto) [reply plugins/notifications-backend/src/database/DatabaseNotificationsStore.ts:176]: @mareklibra I'll close this PR as after following up discussion [here](https://github.com/backstage/backstage/pull/24562) it was decided to implement it slightly differently.

I'll raise soon another PR, and will include there better DB coverage

**Comment by acierto** [comment]
Closing this implementation in favour of https://github.com/backstage/backstage/pull/25015 based on the decision in RFC https://github.com/backstage/backstage/pull/24562

**Pull request closed without merging**